### PR TITLE
SDAIA-13 Wordpress Dataloader API

### PR DESCRIPTION
**Pull request by hinakhadim**
## Description

- This PR contains `Wordpress DataLoader API` which is used to sync data with Wordpress. 
- Also Keep a Record of Instructors Assigned on the WP end.

**Dataloader API:** `{DISCOVERY_URL}/api/v1/dataloader_app/dataloader/`
**CourseRun APIs:** `{DISCOVERY_URL}/api/v1/dataloader_app/search/course_runs/`, `{DISCOVERY_URL}/api/v1/dataloader_app/course_runs/`

Flow: When `Publish` button of course is pressed in Wordpress, `Wordpress` hit the API to dataloader which triggers the `data loader task`. The `dataloader task` hit the API to `wordpress` for getting courses data. The courses data is then updated into the corresponding `course_runs`. Moreover, It also creates or updates the `Subject` (`category` in Wordpress) and `People`(`Instructor` in Wordpress) data.

## Supporting information

JIRA Link: https://edlyio.atlassian.net/browse/SDAIA-13

## Configurations/Additional Information

- `wordpress` related plugins should be installed.
- `tutor plugin common` should be installed with the following config fields.
```python
# discovery-production-settings

WORDPRESS_APP_AUTH_USERNAME='{{ WORDPRESS_ADMIN_USER }}'
WORDPRESS_APP_AUTH_PASSWORD='{{ WORDPRESS_ADMIN_PASSWORD }}'
CELERY_BROKER_URL='{{ CELERY_BROKER_URL }}'
DEFAULT_PRODUCT_SOURCE_SLUG='{{ DISCOVERY_PRODUCT_SOURCE_SLUG }}' (Required for CourseDataLoader API)
```
- Add the following configs in your `tutor/config.yml`
```yaml
CELERY_BROKER_URL: redis://@redis:6379
DISCOVERY_PRODUCT_SOURCE_SLUG: sdaia
```
- Add `sdaia` source in discovery django admin `http://{DISCOVERY_URL}/admin/course_metadata/source/`. *It should match with the `DISCOVERY_PRODUCT_SOURCE_SLUG` entered above in `config.yml`*
- Add `http://wordpress.local.overhang.io/wp-json/` in `Marketing Site API URL` field of `Partner` model in Course Discovery django admin.
- Lastly, Celery worker of discovery should be run for syncing data from wordpress to discovery. Cmd: `python -m celery -A course_discovery worker -l info`

---------------------------------------------------------------------
Other Important Info related to `fetch courses` API for wordpress:
1. `profile_image_url` and `marketing_url` of Instructor is mapped to `major_works` field as json object.
2. `permalink` of Wordpress Category is mapped to `banner_image_url` field of Subject.
3. `featured` of wordpress Course is mapped to `course_overridden` field of Course Run.

## Testing instructions

N/A

**Merge checklist:**
- [x] At least one reviewers should approve.
- [x] Documentation updated (not only docstrings) if needed to.
- [ ] Translations are updated.
- [x] Latest pull of `sdaia-dev` branch before merge after reviewing.
- [ ] Commits are (reasonably) squashed.

**Post merge:**
- [ ] Delete working branch (except for develop and master).

**Screenshots:**
- N/A

## Other information

- This change will not affect course-discovery already working functionality.
- If there is `multi_email_field` error, then `pip install django-multi-email-field`
- If there is some xyz field doesn't exist error on container start, then do `python manage.py migrate`

### CODE REVIEW

**Review by Ali-Salman29 — APPROVED** [review]

**Comment by Ali-Salman29** [comment]
The code looks good to me, however, some tests are failing. Please have a look before merging the changes. 
Also, squash the commits to a single commit.

**Review by khawaja-awais — APPROVED** [review]

File: course_discovery/settings/assets.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from .base import *
+
+COMPRESS_ENABLED = True
+COMPRESS_OFFLINE = True
+
+# Get rid of the "local" handler
+LOGGING["handlers"].pop("local")
+for logger in LOGGING["loggers"].values():
+    if "local" in logger["handlers"]:
+        logger["handlers"].remove("local")
```
Comment (Ali-Salman29) [inline course_discovery/settings/assets.py]: Add an empty line in this file.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 2]
```diff
@@ -393,23 +392,23 @@ def _load_ecommerce_data(self):
                 # Process in batches and wait for the result from the futures
                 pagerange = pageranges['course_runs']
                 for future in concurrent.futures.as_completed(
-                    executor.submit(self._request_course_runs, page) for page in pagerange
+                        executor.submit(self._request_course_runs, page) for page in pagerange
```
Comment (Ali-Salman29) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: Reset unnecessary changes in this file i.e. formatting changes

**Review by Ali-Salman29 — APPROVED** [review]
